Ontology scenario

jtdemer opened the pull request:
The ontology was created on 02/27/2014 with the latest version of the Ontology Creator and uses the current Excel file as input: Metadata, which was created in collaboration with all data providers from the GAIA-X4PLCC-AAD project and can now be merged into the main branch.

Signed-off-by: jtdemer <johannes.demer@asc-s.de>

heuerfin reviewed (commented):
Even though I am not listed as reviewer, I think its fine to merge the current state of the scenario ontology

jdsika commented:
@heuerfin what about this? https://github.com/GAIA-X4PLC-AAD/map-and-scenario-data/blob/main/meta-data/ontologies/gaiax4plc/gaiax4plc_meta_auto_Scenario.ttl

heuerfin commented:
Yeah the ontology you linked, is the ontology we derived from the excel. So the linked ontology would probably fit better. However, I think this ontology and shacle where intended to include the setlevel meta data attributes according to https://gitlab.setlevel.de/open/processes_and_traceability/traceability_data/-/blob/main/SETLevel_SRMD_and_classifications_for_metadata.pdf?ref_type=heads. I assumed, that @jtdemer wanted to first include that one?

lenasauermann commented:
General Ontology should be linked

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+scenario:scenarioShape a sh:NodeShape ;
```
Shapenames should be in PascalCase
`scenario:ScenarioShape`

Please apply this to the other shapes as well if applicable

robertschubert replied:
unresolve: it is still scenarioShape and not ScenarioShape

rcrswld replied:
fixed

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:minCount 1 ;
+            sh:node scenario:EnvironmentalShape ;
+            sh:order 2 ;
+            sh:path scenario:Environmental ],
```
Attributes should be in camelCase
--> `sh:path scenario:environmental`

Please apply this to the other shapes as well if applicable

rcrswld replied:
fixed

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:path scenario:Environmental ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node scenario:GeneralShape ;
```
There is also a "general:GeneralShape" defined by Mirco (#27)
This one seems to be something different. Is "General" the right name for it?

As I understood Mirco all DataShapes (like HdMap) should include the general:GeneralShape. Did you forget it to include?

rcrswld replied:
Yes, this is scenario-related general content. As this definitions are originated from SRMD / SETlevel-semantics, I would propose to keep it here

rcrswld replied:
To prevent ambiguities, we could rename it to ScenarioGeneralShape. Would this help?

rcrswld replied:
As aligned, will be renamed

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:path scenario:heightSystem ] ;
+    sh:targetClass scenario:GeodeticReferenceSystem .
+
+scenario:BoundingBoxShape a sh:NodeShape ;
```
I have seen the BoundingBoxShape and other Shapes in this file in several other domains like hdmap, too. 
Are they all the same? If yes we should centralize them.
Disadvantage: creating claims the the SD-Creation-Wizard will not be that easy anymore because the Wizard will not know it. The Catalogue does because it loads all Shapes when validating

rcrswld replied:
I have created an issue for that, s. https://github.com/GAIA-X4PLC-AAD/ontology-management-base/issues/34
@3Dbastian will take a look into this topic. I would also prefer to extract those contents into a separate shape which can be used by multiple data categories. Might be also of advantage for the data search later, @lenasauermann.

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:description "EPSG code of the map"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of coordinateSystem failed!"@en ;
+            sh:name "coordinateSystem"@en ;
```
can you provide an `sh:in` to define valid values?

rcrswld replied:
Have done it, but more for demonstration purposes. Generally speaking, it might make more sense referring to the offical specification, e.g. https://epsg.org/home.html
Maybe @3Dbastian can help here out?

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:name "region"@en ;
+            sh:order 2 ;
+            sh:path scenario:region ],
+        [ skos:example "Germany, GER" ;
```
cant you use the address shape which is already in the gaia-x vocabulary?

<http://www.w3.org/2006/vcard/ns#Address>

if yes you can add it as a shape with following approach:
```
[ sh:minCount 1 ;
            sh:node <http://www.w3.org/2006/vcard/ns#Address> ;
            sh:path environmentmodel:address;
            sh:order 5 ] ;
```
Or you could use PostalAddressShape:
```
[ sh:minCount 1 ;
            sh:node gx:PostalAddressShape ;
            sh:path environmentmodel:address;
            sh:order 5 ] ;
```
But this has only one property for a  ISO 3166-2 based string.

rcrswld replied:
Would prefere a reference to ISO 3166-2 (e.g., for Germany ISO 3166-2:DE -> DE-BY). Therefore, we can also refer to gx-shape, right?

rcrswld replied:
Should be solved in the georef-shape. @3Dbastian, please take a look at it.

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:path scenario:relationOrArea ] ;
+    sh:targetClass scenario:ProjectLocation .      
+
+scenario:Data_SourcesShape a sh:NodeShape ;
```
should be DataSourcesShape

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:description "Pegasus type of scenario"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of type failed!"@en ;
+            sh:name "type"@en ;
```
can you add an `sh:in` here?

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:description "Maneuver type in the sense of a functional scenario"@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of maneuver failed!"@en ;
+            sh:name "maneuver"@en ;
```
can you add an `sh:in` here?

rcrswld replied:
The maneuver-properties should be handle with OpenLABEL. Therefore, I would suggest to leave it out here.

robertschubert replied:
ok, resolved

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:description "Purpose of this scenario."@en ;
+            sh:maxCount 1 ;
+            sh:message "Validation of aim failed!"@en ;
+            sh:minCount 0 ;
```
can you add an `sh:in` here?

rcrswld replied:
No standardized enums available

robertschubert replied:
okay, thread resolved

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+            sh:maxCount 1 ;
+            sh:message "Validation of usedStandardFunctions failed!"@en ;
+            sh:minCount 0 ;
+            sh:name "usedStandardFunctions"@en ;
```
can you add an `sh:in` here?

rcrswld replied:
There are a lot of action in OpenSCENARIO. I would suggest to reference to OpenSCENARIO-standard, refering to actions (https://publications.pages.asam.net/standards/ASAM_OpenSCENARIO/ASAM_OpenSCENARIO_XML/latest/generated/content/Action.html). Does that get to the essence of this property, @heuerfin?

jdsika commented:
@rcrswld please take this pull request as your personal responsibility. @robertschubert offers his help to understand how a general ontology from GaiaX is to be understood. 
Especially make sure to consolidate the existing different proposals for a scenario ontology which arwe found in the different repos and excel lists. A target for a merge should be the end of next week.

rcrswld commented:
Alignment from AP 1.4, 17.05.2024:
Delete parts from the existing shape which are overlapping with OpenLABEL. The remaining parts can be published.

robertschubert commented on Scenario/scenario_ontology.ttl:
```diff
@@ -0,0 +1,17 @@
+@prefix gax-core: <https://w3id.org/gaia-x/core#> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix scenario: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/> .
```
Folder is "Scenario" and should be "scenario"

heuerfin replied:
The ontology is moved to `scenario`, I think the shacl shape also, right?

robertschubert replied:
exactly, all files should be located in scenario. it is the case now so this thread is resolved now.

robertschubert commented on Scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,439 @@
+@prefix scenario: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
```
unresolve, the gx prefix is still missing

@prefix gx: https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework# .

rcrswld replied:
gx prefix now added

robertschubert commented:
@rcrswld 
i unresolved ("unresolve:" in comment) two items since I cannot see the change in the repository. Or is only a commit & push missing?

heuerfin commented:
@robertschubert I have a question regarding the subclassing of scenarios. Our concept is to describe a scenario in the following order:
1. Official Standard: OpenLABEL -> Provides an ontology for scenarios. E.g. has the class "Scenario" https://openlabel.asam.net/V1-0-0/ontologies/openlabel_ontology_scenario_tags.ttl
2. Established "standards": e.g. of the SETLevel project : https://gitlab.setlevel.de/open/processes_and_traceability/traceability_data/-/blob/main/SETLevel_SRMD_and_classifications_for_metadata.pdf?ref_type=heads -> currently no formal ontology available in ttl, I've created a very basic example in https://github.com/GAIA-X4PLC-AAD/map-and-scenario-data/blob/main/meta-data/ontologies/setlevel/setlevel.ttl (requires a lot of refactoring but there is at least something). 
3. Project specific metadata

So my basic idea was to use the existing OpenLABEL ontology as a basis and inherit it in our customization. However, the `Scenario` class needs to be a sub-type of `gaia-x:Resource`. In https://www.scitepress.org/Papers/2021/106842/106842.pdf I found that the rdfs standard allows multiple inheritance. So I've tried it with the following example:
```
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix gax-core: <https://w3id.org/gaia-x/core#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix scenario: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

scenario: a owl:Ontology ;
    rdfs:label "Ontology definition for scenario"@en ;
    dcterms:contributor "Fin Heuer (DLR), Rico Auerswald (FhG IVI)" ;
    dcterms:creator "The GAIA-X 4 PLC AAD Project Team" ;
    dcterms:identifier "https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/" ;
    owl:versionIRI <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/Ontology_Scenario/scenario/scenario_ontology.ttl> ;
    owl:versionInfo "0.1"^^xsd:float ;
    owl:imports <https://openlabel.asam.net/V1-0-0/ontologies/openlabel_ontology_scenario_tags.ttl> .

scenario:Scenario a owl:Class ;
    rdfs:label "Scenario" ;
    rdfs:comment "Class definition for a scenario"@en ;
    rdfs:subClassOf gax-core:Resource ;
    rdfs:subClassOf <https://openlabel.asam.net/V1-0-0/ontologies/Scenario> .
```

This seems to work in protege and looks also good. However, I'm not sure if this is the correct way and it will be understood by the catalogue or whatever component will use it. 
I've also created another sample, that looks like this:
```
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix gax-core: <https://w3id.org/gaia-x/core#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix scenario: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix openlabel: <https://openlabel.asam.net/V1-0-0/ontologies/> .

scenario: a owl:Ontology ;
    rdfs:label "Ontology definition for scenario"@en ;
    dcterms:contributor "Fin Heuer (DLR), Rico Auerswald (FhG IVI)" ;
    dcterms:creator "The GAIA-X 4 PLC AAD Project Team" ;
    dcterms:identifier "https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/" ;
    owl:versionIRI <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/Ontology_Scenario/scenario/scenario_ontology.ttl> ;
    owl:versionInfo "0.1"^^xsd:float .

scenario:Scenario a owl:Class ;
    rdfs:label "GXScenario" ;
    rdfs:comment "Class definition for a scenario"@en ;
    rdfs:subClassOf gax-core:Resource ;
    rdfs:subClassOf openlabel:Scenario .
```
It doesn't use the `owl:imports` and directly references the prefixes. And I think regarding the linking it would probably be the correct one.

I didn't wan't to push anything since it will probably break the current changes/discussions. Is something in that direction an option? And @rcrswld would it be also fine for you?

robertschubert commented:
Hi @heuerfin 
yes, you are right, RDF supports multi inheritance. 
I corrected the lower ontology slightly and I can load it successfully into the federated catalogue.

```
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix gax-core: <https://w3id.org/gaia-x/core#> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix scenario: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix openlabel: <https://openlabel.asam.net/V1-0-0/ontologies/> .
@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .

scenario: a owl:Ontology ;
    rdfs:label "Ontology definition for scenario"@en ;
    dcterms:contributor "Fin Heuer (DLR), Rico Auerswald (FhG IVI)" ;
    dcterms:creator "The GAIA-X 4 PLC AAD Project Team" ;
    dcterms:identifier "https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/scenario/" ;
    owl:versionIRI <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/Ontology_Scenario/scenario/scenario_ontology.ttl> ;
    owl:versionInfo "0.1"^^xsd:float .

scenario:Scenario a owl:Class ;
    rdfs:label "GXScenario"@en ;
    rdfs:comment "Class definition for a scenario"@en ;
    rdfs:subClassOf gx:DataResource ,
                    openlabel:Scenario .
```

@lenasauermann FYI

heuerfin commented:
That looks great @robertschubert! Also worked in Protege for me.

@rcrswld I think this would be the correct approach right? I'm currently not quiet sure regarding the `rdfs:label "GXScenario"@en ;`. I used a different label from OpenLABEL since at least in protege somehow the correct linking got lost. However, if it's semantically still correct with the `rdfs:label "Scenario"@en ;` we can use that one. What do you think?

rcrswld commented on Scenario/georeference_shacl.ttl:
first draft for a georeference-shape as a starting point created, based on the already existing properties from the scenario-shape

rcrswld commented on Scenario/scenario_shacl.ttl:
I deleted following properties because they should be covered by OpenLABEL:
maneuver
trafficParticipantTypes
environmentConditions
sunElevation
wetness
ruleOfTheRoad

@heuerfin, are you fine with this suggestion?

heuerfin replied:
Yes seems nice. Do we need to use then the `owl:imports` in order to use the OpenLABEL attributes?

rcrswld reviewed (changes requested):
Review finished - fixed issues from robert, extracted some attributes into a general shape for georeferences as draft and deleted attributes which are already covered by OpenLABEL

rcrswld commented:
AP 1.4 alignment from 24.05.:
georef:
- georeference-properties should be deleted out of the scenario shacl - just a reference via 'sh:node'
- SD-wizard is not able to resolve links to other shapes -> therefore, for generation of the example claim, it should be included temporary for the example-claim-creation

general-shape:
- reference to general-namespace should be included

3Dbastian commented:
> AP 1.4 alignment from 24.05.: georef:
> 
>     * georeference-properties should be deleted out of the scenario shacl - just a reference via 'sh:node'
> 
>     * SD-wizard is not able to resolve links to other shapes -> therefore, for generation of the example claim, it should be included temporary for the example-claim-creation
> 
> 
> general-shape:
> 
>     * reference to general-namespace should be included

rcrswld reviewed (approved):
Last aligned issues from 24.05., AP 1.4 meeting has been fixed. Should be fine so far.

rcrswld commented:
@jtdemer as Robert and Justin cannot approve this pull request next week, could you do it instead, please?

robertschubert commented on scenario/scenario_ontology.ttl:
```diff
@@ -0,0 +1,23 @@
+@prefix dcterms: <http://purl.org/dc/terms/> .
+@prefix gax-core: <https://w3id.org/gaia-x/core#> .
+@Prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .
```
prefix should be lowercase

heuerfin replied:
Fixed in [6a622d4](https://github.com/GAIA-X4PLC-AAD/ontology-management-base/commit/6a622d4fdd24480f16758d3943524710e5b06740)

robertschubert commented on CODEOWNERS:
```diff
@@ -11,7 +11,7 @@
 
 # scenario directory (Rico AND Justin)
```
small thing: could you replace Justin with Robert?

heuerfin replied:
Fixed in [6a622d4](https://github.com/GAIA-X4PLC-AAD/ontology-management-base/commit/6a622d4fdd24480f16758d3943524710e5b06740)

robertschubert commented on scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,262 @@
+            sh:path scenario:country ] ;
+    sh:targetClass scenario:Traffic .
+
+scenario:LinkShape a sh:NodeShape ;
```
as far as I can see the LinkShape is already there in `general`? Or is this something different?

heuerfin replied:
This is right, fixed in f19ba90cd1a8a55ce44f67212592d88f69dd00d4

robertschubert commented on scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,262 @@
+            sh:minCount 1 ;
+            sh:node general:GeneralShape ;
+            sh:order 5 ;
+            sh:path general:general ],
```
sorry, my fault. 
sh:path should be `scenario:general`
I will add a section to the readme.

heuerfin replied:
Fixed in f19ba90cd1a8a55ce44f67212592d88f69dd00d4

heuerfin commented:
> AP 1.4 alignment from 24.05.: georef:
> 
> * georeference-properties should be deleted out of the scenario shacl - just a reference via 'sh:node'
> * SD-wizard is not able to resolve links to other shapes -> therefore, for generation of the example claim, it should be included temporary for the example-claim-creation
> * georef-shape in the scenario-folder should now be deleted as a separate shape in geo-ref folder is existing
> 
> general-shape:
> 
> * reference to general-namespace should be included
> 
> example-claim:
> 
> * still open, @heuerfin would you please try a first draft?

I've added an example claim in https://github.com/GAIA-X4PLC-AAD/scenario-asset-example/blob/main/Metadata/ALKS_Scenario_4.4_1_CutInNoCollision_TEMPLATE_jsonld_claim.json. Since i did not know where to place it in this repository.

As you suggested I added temporarily the general and georeference shape in a combined file. However, the referenced scenario does not have some georeference, so I did not use it.

jtdemer commented:
> > AP 1.4 alignment from 24.05.: georef:
> > 
> > * georeference-properties should be deleted out of the scenario shacl - just a reference via 'sh:node'
> > * SD-wizard is not able to resolve links to other shapes -> therefore, for generation of the example claim, it should be included temporary for the example-claim-creation
> > * georef-shape in the scenario-folder should now be deleted as a separate shape in geo-ref folder is existing
> > 
> > general-shape:
> > 
> > * reference to general-namespace should be included
> > 
> > example-claim:
> > 
> > * still open, @heuerfin would you please try a first draft?
> 
> I've added an example claim in https://github.com/GAIA-X4PLC-AAD/scenario-asset-example/blob/main/Metadata/ALKS_Scenario_4.4_1_CutInNoCollision_TEMPLATE_jsonld_claim.json. Since i did not know where to place it in this repository.
> 
> As you suggested I added temporarily the general and georeference shape in a combined file. However, the referenced scenario does not have some georeference, so I did not use it.

Hi @heuerfin, 
You can place a correctly typed sample directly into the `scenario` folder, as I have done it here for the environmental model: https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/environment-model/environment-model/EnvironmentModel_instance.json The content should be logical but it doesn’t need to refer to a specific scenario.

heuerfin commented:
Ok 

> > > AP 1.4 alignment from 24.05.: georef:
> > > 
> > > * georeference-properties should be deleted out of the scenario shacl - just a reference via 'sh:node'
> > > * SD-wizard is not able to resolve links to other shapes -> therefore, for generation of the example claim, it should be included temporary for the example-claim-creation
> > > * georef-shape in the scenario-folder should now be deleted as a separate shape in geo-ref folder is existing
> > > 
> > > general-shape:
> > > 
> > > * reference to general-namespace should be included
> > > 
> > > example-claim:
> > > 
> > > * still open, @heuerfin would you please try a first draft?
> > 
> > 
> > I've added an example claim in https://github.com/GAIA-X4PLC-AAD/scenario-asset-example/blob/main/Metadata/ALKS_Scenario_4.4_1_CutInNoCollision_TEMPLATE_jsonld_claim.json. Since i did not know where to place it in this repository.
> > As you suggested I added temporarily the general and georeference shape in a combined file. However, the referenced scenario does not have some georeference, so I did not use it.
> 
> Hi @heuerfin, You can place a correctly typed sample directly into the `scenario` folder, as I have done it here for the environmental model: https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/environment-model/environment-model/EnvironmentModel_instance.json The content should be logical but it doesn’t need to refer to a specific scenario.

Ok then it would be better if its a "full" sample with all attributes instead of matching ones, right? I've added one using all attributes in https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/Ontology_Scenario/scenario/Scenario_instance.json. However, wouldn't it make sense that we use lower case names in the file since it would match now the directory and file names of the ontology and shacl?

jdsika commented:
Lower case please :)

heuerfin commented:
Done.

jdsika commented:
@heuerfin you should in general compare with the [guidelines](https://github.com/GAIA-X4PLC-AAD/ontology-management-base?tab=readme-ov-file#modeling-guidelines) and also give feedback if those need changes additions as well, please

heuerfin commented:
Thanks for the remark! I think it complies now with these guidelines. And I have seen that Johannes has also just changed this in accordance with these guidelines.

jdsika commented:
@heuerfin good!

What is still needed for a merge here? Who feels confident enough to give it a technical approval regarding the content?

heuerfin commented:
I think there are two more comments from @robertschubert open where im not entirly sure whats the correct approach. @rcrswld can you handle these?
And I'm not sure if #38 needs an adoption on the ontology. However, we can probably do it afterwards.

jtdemer commented:
@lenasauermann as discussed in the dataprovider meeting today, please approve the pull request on behalf of @robertschubert

lenasauermann commented on scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,247 @@
+            sh:name "customCommands"@en ;
+            sh:order 6 ;
+            sh:path scenario:customCommands ] ;
+    sh:targetClass scenario:Participants .
```
"Participants" seems like a unwise naming as we already have participants as entity participating in a data ecosystems. 

Would a renaming into "Objects" make sense?

heuerfin replied:
Yes this could lead to some confusion. However, `Objects` would refer also in the context of a scenario to anything also static things like a tree or so. I'd like to stick somehow to something with participants, would `trafficParticipants` would be alright for you?

lenasauermann replied:
@MircoNierenz Fin already adjusted the shacl

lenasauermann commented on scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,247 @@
+            sh:node general:LinkShape ;
+            sh:order 2 ;
+            sh:path scenario:trafficSpace ] ;
+    sh:targetClass scenario:Structural .
```
Aren't those links covered by related or required data in general:General?

heuerfin replied:
Yeah this is definetly covered. It also becomes clearer with some improvements in the use of the required links. However, in a scenario there is a pretty tight coupling between the scenario and the map. I think with the general links you would have to really inspect each link to know which map this scenario refers to. Having a direct attribute would resemble the strong coupling and I think also reduce the lookup effort. 

So if it's not too bad, I'd like to have both.

lenasauermann replied:
Okay, then let's have both

heuerfin commented on scenario/scenario_instance.json:
```diff
@@ -0,0 +1,275 @@
+    "scenario:timeDate": {
+      "@value": "2024-05-27T10:00:00",
+      "@type": "xsd:dateTime"
+    },
```
Hm... I don't think so. This refers to the `scenario:ScenarioGeneralShape`, but I'm not 100% sure. This was generated by the wizard and I think it's fine. However, we should probably rename the ScenarioGeneral stuff into else, like `scenario:Description` or so?

MircoNierenz replied:
Description fits better than general - I have adjusted it in the excel table

lenasauermann replied:
Fin and I changed it to "common"

lenasauermann replied:
Fin and I changed it to "common"

heuerfin commented on scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,247 @@
+            sh:message "Validation of type failed!"@en ;
+            sh:name "type"@en ;
+            sh:order 0 ;
+            sh:path scenario:type ],
```
Yeah this is also quiet confusing. We should probably renmane the https://github.com/GAIA-X4PLC-AAD/ontology-management-base/blob/Ontology_Scenario/scenario/scenario_shacl.ttl#L81C7-L89C35 into `dataFormat`. And the other type into `abstractionLevel`. I think this will then be more clear.

MircoNierenz replied:
for the generic creation of such shacl's I would rather include the subcategory name, i.e.
scenarioFormatType
scenarioGeneralType

lenasauermann replied:
Fin changed it to dataFormat and abstractionLevel. Including the subcategory name is not necesarry as it is "scenario:dataFormat"

heuerfin commented on scenario/scenario_shacl.ttl:
```diff
@@ -0,0 +1,247 @@
+            sh:name "usedStandardFunctions"@en ;
+            sh:order 3 ;
+            sh:path scenario:usedStandardFunctions ] ;
+    sh:targetClass scenario:General .
```
Ok yes this fits my understanding from above. These are attributes that shouldn't be integrated into `general:General` but these are somehow general attributes for a scenario. Would `Description` would be more suitable?

MircoNierenz replied:
Description fits better than general - I have adjusted it in the excel table

lenasauermann replied:
Fin and I changed it to "common"

lenasauermann commented on scenario/scenario_instance.json:
```diff
@@ -0,0 +1,275 @@
+      "@type": "general:Links",
+      "general:data": {
+        "@type": "general:Link",
+        "scenario:url": {
```
When type "general:Links" is used, also url and type needs to be from namespace general as they are defined there

lenasauermann replied:
I only mark it here, but it appears a few times

lenasauermann replied:
@heuerfin can you change this?

heuerfin replied:
Fixed in b618f4e

lenasauermann commented on scenario/scenario_instance.json:
```diff
@@ -0,0 +1,275 @@
+      "@type": "xsd:string"
+    }
+  },
+  "general:general": {
```
scenario:general and not general:general

lenasauermann replied:
@heuerfin can you change this?

MircoNierenz replied:
done

heuerfin reviewed (approved):
All changes implemented in d0b8a4d. Also coordinated with mirco (deletion of unused gx prefix). Since it is now complete, this branch can be merged.

robertschubert commented:
@heuerfin @MircoNierenz 
after rebase this PR can be merged.

robertschubert commented:
merge from main done, VARIABLE.md existent.
Merge to main.
